### ISSUE TITLE
[Question] Enable underline

### ISSUE THREAD

**Issue by karambaq** [zdharma/fast-syntax-highlighting#17]
Hello, great plugin!
How can I enable underline like in the standart syntax-highlighting plugin?
Thanks.

**Comment by psprint** [comment]
No it's removed for speed, but I see now that peek performance might be not worth removing that, I may restore it maybe.

**Comment by simeg** [comment]
This is the only reason from stopping me from using this plugin. I like all features it offers but I don't think it's clear when I've resolved a folder. The folder turns pinkish, which I first thought was something bad (like red). It would be awesome if this could be configurable!

Or if you could aid us in some guidance to where in the codebase the changes would need to be, and someone else could implement it. Thanks.

**Comment by psprint** [comment]
Underline was to highlight if part of filename was OK. So if there's folder `directory`, and one enters `ls dire` then `dire` is underlined. I cannot recall what happened when one entered whole path, i.e. `directory`, f-sy-h highlights in pink, because IMO default z-sy-h reaction was too subtle, and I often missed it, I remember that. The same for removing underline, iTerm2 showed so thin line, that I never noticed that z-sy-h is tracking parts of filenames. Is your request about restoring underline for `ls dire`, etc., or to change magenta to other color? Because magenta can be configured.

**Comment by simeg** [comment]
z-sy-h shows without underline if nothing is matched, but if something is matched then it shows underline (even if the whole path isn't matched). So it gives you a hint if you're on track to something. Maybe this is the behavior for this plugin too, can't remember the details now.

Oh, the color is configurable - cool. Then my request is about making it configurable to add underline for matched files / dirs.

**Comment by psprint** [comment]
> So it gives you a hint if you're on track to something.

Yes, it's an interesting feature, I might restore it to f-sy-h, performance drop caused by globbing the parts of filenames wasn't too big. Unless... you're on not very fast network drive like NFS or SSHFS. But this can be solved by adding array `FAST_DONT_GLOB_PATHS=( "/mount/mynfs1/\*" "$HOME/my_sshfs_to_NAS/\*" ), i.e. a blacklist of directories where the underline is deactivated.

**Comment by simeg** [comment]
If it could be configurable I guess everyone would be happy, but I'm not sure how you usually configure zsh plugins that are not executables. However if the core focus of the plugin is speed then maybe it doesn't make sense to have it underline matched dirs/files.

**Comment by leotaku** [comment]
Are you still interested in pursuing this?

**Comment by psprint** [comment]
Ah, yes, thanks for recalling me it
